src/server.py

```diff
@@ -104,4 +104,155 @@ async def discard_cards(deck_id: str, cards: str):
     
     return {"message": "Cards discarded successfully"}
 
+
+
+@app.get("/api/v2/highestRank/~{cards}")
+async def highest_rank(cards: str):
+    hand = stringToCards(cards)
+    if isStraightFlush(hand):
+        return {"message": "Straight Flush"}
+    if isFourOfAKind(hand):
+        return {"message": "Four of A Kind"}
+    if isFullHouse(hand):
+        return {"message": "Full House"}
+    if isFlush(hand):
+        return {"message": "Flush"}
+    if isStraight(hand):
+        return {"message": "Straight"}
+    if isThreeOfAKind(hand):
+        return {"message": "Three of A Kind"}
+    if isTwoPair(hand):
+        return {"message": "Two Pair"}
+    if isPair(hand):
+        return {"message": "One Pair"}
+    
+    return {"message": "High Card"}
+
+def isStraightFlush(hand: list[Card]) -> bool:
+    isstraight = isStraight(hand)
+    isflush = isFlush(hand)
+    return isstraight and isflush
+
+def isFlush(hand: list[Card]) -> bool:
+    flush = True
+    suit = hand[0].getSuit() 
+    for c in hand:
+        if c.getSuit() != suit:
+            flush = False
+            break
+    
+    return flush
+
+# a true abomination of coding style but I think it will work           
+def isStraight(hand: list[Card]) -> bool:
```

Comment:
This would be a good opportunity to write a test, but we haven't gotten to that part of the class yet :)